File: samples/jvm/src/main/java/dev/pubgkt/MatchesSample.java
package dev.pubgkt;

import java.util.List;

import kotlin.coroutines.EmptyCoroutineContext;
import kotlinx.coroutines.BuildersKt;

import dev.pubgkt.matches.GetMatchByIdKt;
import dev.pubgkt.matches.Match;
import dev.pubgkt.players.GetPlayersByNameKt;
import dev.pubgkt.players.Player;
import dev.pubgkt.players.PlayerMatch;

public class MatchesSample {
    public static void main(String[] args) throws InterruptedException {
        if (args.length == 0) {
            throw new IllegalArgumentException("Usage: MatchesSample <api-key>");
        }

        PubgApi api = new PubgApi(args[0]);

        System.out.println("=== getMatchById ===");
        List<Player> players = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> GetPlayersByNameKt.getPlayersByNames(api, List.of("sparkingg"), Platform.STEAM, cont)
        );
        List<PlayerMatch> matches = players.get(0).getMatches();
        String matchId = matches.get(0).getId();

        Match match = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> GetMatchByIdKt.getMatchById(api, matchId, Platform.STEAM, cont)
        );
        System.out.println("Match: " + match.getId());
        System.out.println("Map: " + match.getMapName() + ", Mode: " + match.getGameMode() + ", Duration: " + match.getDuration() + "s");
        System.out.println("Participants: " + match.getParticipants().size() + " teams");
    }
}

File: samples/jvm/src/main/kotlin/dev/pubgkt/MatchesSample.kt
package dev.pubgkt

import dev.pubgkt.matches.getMatchById
import dev.pubgkt.players.getPlayersByNames

suspend fun main(args: Array<String>) {
    val apiKey = args.getOrNull(0) ?: error("Usage: MatchesSample <api-key>")
    val api = PubgApi(apiKey)

    println("=== getMatchById ===")
    val players = api.getPlayersByNames("sparkingg")
    val matchId = players.first().matches.firstOrNull()?.id ?: error("Player has no recent matches")
    val match = api.getMatchById(matchId)
    println("Match: ${match.id}")
    println("Map: ${match.mapName}, Mode: ${match.gameMode}, Duration: ${match.duration}s")
    println("Participants: ${match.participants.size} teams")
    match.participants.take(3).forEach { team ->
        println("  Team ${team.teamId} (rank ${team.rank}):")
        team.players.forEach { p -> println("    ${p.name} — ${p.kills} kills, ${p.damageDealt} dmg") }
    }
}
